lib/metric.go

```diff
@@ -172,26 +165,38 @@ func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
 	}
 }
 
-func main() {
-	runtime.GOMAXPROCS(2)
-	if len(os.Args) != 2 {
-		log.Fatalf("Usage: %s METRICS_URL", os.Args[0])
+func (f *Family) AddLabel(key, val string) {
+	for i, item := range f.Metrics {
+		switch item.(type) {
+		case Metric:
+			m := item.(Metric)
+			m.Labels[key] = val
+			f.Metrics[i] = m
+		}
 	}
+}
 
-	mfChan := make(chan *dto.MetricFamily, 1024)
-
-	go fetchMetricFamilies(os.Args[1], mfChan)
-
-	result := []*metricFamily{}
-	for mf := range mfChan {
-		result = append(result, newMetricFamily(mf))
-	}
-	json, err := json.Marshal(result)
-	if err != nil {
-		log.Fatalln("error marshaling JSON:", err)
-	}
-	if _, err := os.Stdout.Write(json); err != nil {
-		log.Fatalln("error writing to stdout:", err)
+func (f *Family) ToOpenTSDBv1() string {
```

Comment:
Let's keep a refactoring separate from an added feature.

I think there is more to say about this, as OpenTSDB has a more restricted character set for label values.